storage_scrubber/src/pageserver_physical_gc.rs

```diff
@@ -597,6 +602,60 @@ async fn gc_timeline(
         }
     };
 
+    if let Some(tenant_manifest_info) = &*tenant_manifest_info {
+        // TODO: this is O(n^2) in the number of offloaded timelines. Do a hashmap lookup instead.
+        let maybe_offloaded = tenant_manifest_info
+            .manifest
+            .offloaded_timelines
+            .iter()
+            .find(|offloaded_timeline| offloaded_timeline.timeline_id == ttid.timeline_id);
+        if let Some(offloaded) = maybe_offloaded {
+            let warnings = validate_index_part_with_offloaded(index_part, offloaded);
+            let warn = if warnings.is_empty() {
+                false
+            } else {
+                // Verify that the manifest hasn't changed. If it has, a potential racing change could have been cause for our troubles.
+                match list_tenant_manifests(remote_client, ttid.tenant_shard_id, target).await? {
```

Comment:
A more precise comment on the race condition will surely appreciated by future readers.
Sth like "manifest is always updated after timeline's index_part.json", but scrubber reads it in the other order".

Reply:
> manifest is always updated after timeline's index_part.json", but scrubber reads it in the other order

former isn't true, during an unarchival operation where we also unoffload a timeline, we do:

1. unoffload_timeline, during which:
  a) load index_part.json to memory
  b) create Timeline object via `load_remote_timeline`
  c) write new manifest to S3 with timeline removed from the offloaded timelines list
2.  set `archived_at` to null in `index_part.json`, upload it to S3

but yes, for the archival + offloading case, we first write the index_part with new `archived_at`, then add it to the offloaded timelines list.